microsoft-edge/webview2/how-to/context-menus.md

```diff
@@ -0,0 +1,911 @@
+                kind == COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_RADIO)
+            {
+                if (isEnabled)
+                {
+                    if (isChecked)
+                    {
+                        CHECK_BOOL(AppendMenu(hPopupMenu, 
+                                              MF_CHECKED | MF_STRING, 
+                                              commandId, 
+                                              labelString.c_str()));
+                    }
+                    else
+                    {
+                        CHECK_BOOL(AppendMenu(hPopupMenu, 
+                                              MF_BYPOSITION | MF_STRING,
+                                              commandId, 
+                                              labelString.c_str()));
+                    }
+                }
+                else
+                {
+                    if (isChecked)
+                    {
+                        CHECK_BOOL(AppendMenu(hPopupMenu, 
+                                              MF_CHECKED | MF_GRAYED | MF_STRING, 
+                                              commandId, 
+                                              labelString.c_str()));
+                    }
+                    else
+                    {
+                        CHECK_BOOL(AppendMenu(hPopupMenu, 
+                                              MF_GRAYED | MF_STRING, 
+                                              commandId, 
+                                              labelString.c_str()));
+                    }
+                }
+            }
+            else if (kind == COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_COMMAND)
+            {
+                if (isEnabled)
+                {
+                    CHECK_BOOL(AppendMenu(hPopupMenu,
+                                          MF_BYPOSITION | MF_STRING, 
+                                          commandId, 
+                                          labelString.c_str()));
+                }
+                else
+                {
+                    CHECK_BOOL(AppendMenu(hPopupMenu, 
+                                          MF_GRAYED | MF_STRING, 
+                                          commandId, 
+                                          labelString.c_str()));
+                }
+            }
+        }
+    }
+```
+
+---
+
+
+<!-- ====================================================================== -->
+## Adding menu items to a context menu
+
+
+<!-- -------------------------------------------------- -->
+### Adding default menu items to a custom context menu
+
+Each menu item is of a specific kind.
+
+
+# [C#](#tab/csharp)
+
+On a `CoreWebView2.ContextMenuRequested` event, add an event listener that has a `CoreWebView2ContextMenuRequestedEventArgs`.
+
+Iterate through an `IList` of `CoreWebView2ContextMenuItem` items, adding a `CoreWebView2ContextMenuItem` for each menu item.  Test the `.Kind` of each menu item, such as `Separator` or `Command`.
```

Comment:
Something like -  `MenuItems` provides the tree of the WebView2's context menu items for the right-clicked context. Iterate through the `IList<CoreWebView2ContextMenuItem>` to include WebView2 context menu items in your app's context menu?

Reply:
Please try clicking [+/-] (Suggestion) to enter a Markdown-tagged revision here.

Which API item is "`- MenuItems`", in your comment above?     @yildirimcagri 

Do you want me to add your sentence below "Iterate through...", or replace the "Iterate through" sentence by your sentence?

Reply:
Resolved, 
`The `MenuItems` property of `CoreWebView2ContextMenuRequestedEventArgs` provides the tree of the WebView2's context menu items for the right-clicked context.  To include WebView2 context menu items in your app's context menu, iterate through the `IList<CoreWebView2ContextMenuItem>`, adding a `CoreWebView2ContextMenuItem` for each menu item.  Test the `.Kind` of each menu item, such as `Separator` or `Command`.`